packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -78,14 +91,20 @@ class DatePickerInput extends React.Component {
     this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
   }
 
-  handleOnButtonClick() {
+  componentDidUpdate(prevProps) {
+    if (this.props.shouldShowPicker && !prevProps.shouldShowPicker && this.props.onClick) {
+      this.props.onClick();
+    }
+  }
+
+  handleOnButtonClick(event) {
     // The picker is about to display so request focus from the containing component (e.g. modal) if it has the focus trapped.
     if (this.props.requestFocus) {
       this.props.requestFocus();
     }
 
-    if (this.props.onClick) {
-      this.props.onClick();
+    if (this.props.onCalendarButtonClick) {
+      this.props.onCalendarButtonClick(event, this.props.onClick);
```

Comment:
want to add if check on if this.props.onClick is valid

Reply:
Updated: https://github.com/cerner/terra-core/pull/974/commits/619c1d70775ed668582319dfa0410181bc8e4e55